modules/vdoaiBidAdapter.js

```diff
@@ -31,23 +111,80 @@ export const spec = {
         sizes: sizes,
         bidId: bidRequest.bidId,
-        // TODO: is 'page' the right value here?
-        referer: bidderRequest.refererInfo.page,
-        // TODO: fix auctionId leak: https://github.com/prebid/Prebid.js/issues/9781
-        id: bidRequest.auctionId,
-        mediaType: bidRequest.mediaTypes.video ? 'video' : 'banner'
+        mediaType: bidRequest.mediaTypes.video ? 'video' : 'banner',
+        domain: bidderRequest.ortb2.site.domain,
+        publisherDomain: bidderRequest.ortb2.site.publisher.domain,
+        adUnitCode: bidRequest.adUnitCode,
+        bidder: bidRequest.bidder,
+        tmax: bidderRequest.timeout
       };
+
+      payload.bidderRequestId = bidRequest.bidderRequestId;
+      payload.auctionId = deepAccess(bidRequest, 'ortb2.source.tid');
+      payload.transactionId = deepAccess(bidRequest, 'ortb2Imp.ext.tid');
+      payload.gpid = deepAccess(bidRequest, 'ortb2Imp.ext.gpid') || deepAccess(bidRequest, 'ortb2Imp.ext.data.pbadslot');
+      payload.ortb2Imp = deepAccess(bidRequest, 'ortb2Imp');
+
+      if (payload.mediaType === 'video') {
+        payload.context = bidRequest.mediaTypes.video.context;
+        payload.playerSize = parseVideoSize(bidRequest);
+        payload.mediaTypeInfo = deepClone(bidRequest.mediaTypes.video);
+      }
+
+      if (typeof bidRequest.getFloor === 'function') {
+        let floor = bidRequest.getFloor({
+          currency: 'USD',
+          mediaType: '*',
+          size: '*'
+        });
+        if (floor && floor.floor && floor.currency === 'USD') {
+          payload.bidFloor = floor.floor;
+        }
+      } else if (bidRequest.params.bidFloor) {
+        payload.bidFloor = bidRequest.params.bidFloor;
+      }
+
+      payload.pageInfo = getPageInfo(bidderRequest);
+
+      if (bidderRequest && bidderRequest.gdprConsent) {
+        payload.gdprConsent = {
+          consentRequired: bidderRequest.gdprConsent.gdprApplies,
+          consentString: bidderRequest.gdprConsent.consentString,
+          addtlConsent: bidderRequest.gdprConsent.addtlConsent
+        };
+      }
+      if (bidderRequest && bidderRequest.gppConsent) {
+        payload.gppConsent = {
+          applicableSections: bidderRequest.gppConsent.applicableSections,
+          consentString: bidderRequest.gppConsent.gppString,
+        }
+      }
+      if (bidderRequest && bidderRequest.ortb2) {
+        payload.ortb2 = bidderRequest.ortb2;
+      }
+      if (bidderRequest && bidderRequest.uspConsent) {
+        payload.usPrivacy = bidderRequest.uspConsent;
+      }
+      if (validBidRequests && validBidRequests.length !== 0 && validBidRequests[0].schain && isSchainValid(validBidRequests[0].schain)) {
+        payload.schain = validBidRequests[0].schain;
+      }
+      if (validBidRequests && validBidRequests.length !== 0 && validBidRequests[0].userIdAsEids) {
+        payload.userId = validBidRequests[0].userIdAsEids;
+      }
+      if (config.getConfig('coppa') === true) {
```

Comment:
Can you please access coppa from `bidderRequest.ortb2.regs.coppa` instead of config

Reply:
Hey @ChrisHuie , thanks for the review. we have made the necessary changes. Is there anything else I can assist with?